assembly/src/assembler/mast_forest_builder.rs

```diff
@@ -554,6 +554,21 @@ impl IndexMut<DecoratorId> for MastForestBuilder {
     }
 }
 
+// ------------------------------------------------------------------------------------------------
+
+impl MastForestBuilder {
+    /// Merge an AdviceMap into the one being built within the MAST Forest.
+    pub fn merge_advice_map(&mut self, adv_map: &AdviceMap) -> Result<(), AssemblyError> {
+        match self.mast_forest.advice_map_mut().merge(adv_map) {
+            Some((key, prev_values, new_values)) => Err(AssemblyError::Forest(
+                "AdviceMapKeyCollisionOnMerge",
+                MastForestError::AdviceMapKeyAlreadyPresent { key, prev_values, new_values },
```

Comment:
nit: `AdviceMapKeyCollisionOnMerge` is somewhat inconsistent with `AdviceMapKeyAlreadyPresent`. But also, this will probably change after https://github.com/0xMiden/miden-vm/pull/1881.